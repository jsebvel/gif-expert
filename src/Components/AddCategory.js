import { useState } from "react";
import Proptypes from 'prop-types';

export const AddCategory = ({ setCategories }) => {
    const [inputValue, setInputValue] = useState('');
    const handleInputChange = (e) => {
        setInputValue(e.target.value);
    }

    const handleSubmit = (e) => {
        e.preventDefault();
        if (inputValue.trim().length > 2){
            setCategories(lastCategories => [ inputValue, ...lastCategories]);
            setInputValue('');
        }
    }
    return (
        <form onSubmit={handleSubmit}>
            <input type="text"
                value={inputValue}
                onChange={(e) => {
                    handleInputChange(e);
                }}
            ></input>
        </form>
    )
}

AddCategory.propTypes = {
    setCategories: Proptypes.func.isRequired
}