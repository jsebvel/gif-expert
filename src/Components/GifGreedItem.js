import PropTypes from 'prop-types';
export const GifGreedItem = ({ title, url }) => {
    return (
        <div className="card ">
            <p>{title}</p>
            <img src={url} alt={title} />
        </div>
    )
}

GifGreedItem.propTypes = {
    title: PropTypes.string.isRequired,
    url: PropTypes.string.isRequired
}
    