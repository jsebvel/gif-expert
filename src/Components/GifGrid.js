import { useState, useEffect } from "react";
import { getGifs } from "../helpers/getGifs";
import { useFetchGifs } from "../hooks/useFetchGifs";
import { GifGreedItem } from "./GifGreedItem";
import PropTypes from 'prop-types';


export const GifGrid = ({ category }) => {
    // const [images, setImages] = useState([])
    // useEffect(() => {
    //     getGifs(category).then(imgs => setImages(imgs));
    // }, [category])
    const { data: images, loading } = useFetchGifs(category);

    return (
        <>
            <h3>{category}</h3>
            {loading && <p className="animate_animated animate_fadeIn">{category}</p>}
            <div className="card-grid animate__animated animate__bounce">
                {
                    images.map(img =>
                        <GifGreedItem
                            key={img.id}
                            {...img} />
                    )
                }
            </div>
        </>
    )
}

GifGrid.protoTypes = {
    category: PropTypes.string.isRequired
}
