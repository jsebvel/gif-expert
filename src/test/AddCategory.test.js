import { shallow } from 'enzyme';
import { AddCategory } from '../Components/AddCategory';

describe('test in add category', () => {
    const setCategories = jest.fn();
    let wrapper;
    beforeEach(() => {
        jest.clearAllMocks();
        wrapper = shallow(<AddCategory setCategories={setCategories}></AddCategory>);

    });

    test('snapshop match', () => {
        expect(wrapper).toMatchSnapshot();
    });

    test('Should change input text', () => {
        const inputValue = 'kid vs cat';
        const input = wrapper.find('input');
        input.simulate('change', { target: { value: inputValue } });
    });

    test('Should not call on submit', () => {
        wrapper.find('form').simulate('submit', { preventDefault() { } });
        expect(setCategories).not.toHaveBeenCalled()
    });

    test('should call set categories and clean input text', () => {
        const inputValue = 'catwoman';
        const input = wrapper.find('input').simulate('change', { target: { value: inputValue } });
        const form = wrapper.find('form').simulate('submit', { preventDefault() { } });
        expect(setCategories).toHaveBeenCalled();
        expect(wrapper.find('input').prop('value')).toBe('');

    });
});