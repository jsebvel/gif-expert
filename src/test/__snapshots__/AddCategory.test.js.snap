// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test in add category snapshop match 1`] = `
<form
  onSubmit={[Function]}
>
  <input
    onChange={[Function]}
    type="text"
    value=""
  />
</form>
`;
